RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.cc

```diff
@@ -216,33 +220,110 @@ math::XYZPoint HGCalCLUEAlgo::calculatePosition(const std::vector<int>& v, const
+        for (int yBin = search_box[2]; yBin < search_box[3] + 1; ++yBin) {
+          int binId = lt.getGlobalBinByBin(xBin, yBin);
+          size_t binSize = lt[binId].size();
+
+          for (unsigned int j = 0; j < binSize; j++) {
+            unsigned int otherId = lt[binId][j];
+            if (!cellsOnLayer.isSilic[otherId])
+              continue;  //cells in the silicon cannot talk to cells in scintillator
+            if (distance(i, otherId, layerId, false) < delta_c_) {
+              cellsOnLayer.rho[i] += (i == otherId ? 1.f : 0.5f) * cellsOnLayer.weight[otherId];
+            }
+          }
+        }
+      }
+    } else {
+      delta_c_ = delta_r;
+      std::array<int, 4> search_box = lt.searchBoxEtaPhi(cellsOnLayer.eta[i] - delta_c_,
+                                                         cellsOnLayer.eta[i] + delta_c_,
+                                                         cellsOnLayer.phi[i] - delta_c_,
+                                                         cellsOnLayer.phi[i] + delta_c_);
+      cellsOnLayer.rho[i] += cellsOnLayer.weight[i];
+      float northeast(0), northwest(0), southeast(0), southwest(0), all(0);
+
+      for (int etaBin = search_box[0]; etaBin < search_box[1] + 1; ++etaBin) {
+        for (int phiBin = search_box[2]; phiBin < search_box[3] + 1; ++phiBin) {
+          int binId = lt.getGlobalBinByBinEtaPhi(etaBin, phiBin);
+          size_t binSize = lt[binId].size();
+
+          for (unsigned int j = 0; j < binSize; j++) {
+            unsigned int otherId = lt[binId][j];
+            if (cellsOnLayer.isSilic[otherId])
+              continue;  //cells in the silicon cannot talk to cells in scintillator
+            if (distance(i, otherId, layerId, true) < delta_c_) {
+              // need to round these numbers to 5 decimals to be able to compare floats
+              float roundDec = 100000.;
+              float iPhi = round(cellsOnLayer.phi[i] * roundDec) / roundDec;
+              float otherPhi_ = cellsOnLayer.phi[otherId];
+              otherPhi_ += (otherPhi_ * iPhi >= 0 || abs(iPhi) < 1.) ? 0. : iPhi > 0. ? 2 * M_PI : -2 * M_PI;
+              float otherPhi = round(otherPhi_ * roundDec) / roundDec;
+              float otherEta = round(cellsOnLayer.eta[otherId] * roundDec) / roundDec;
+              float iEta = round(cellsOnLayer.eta[i] * roundDec) / roundDec;
+              LogDebug("HGCalCLUEAlgo") << "  Debugging calculateLocalDensity for Scintillator: \n"
+                                        << "    cell: " << otherId << " energy: " << cellsOnLayer.weight[otherId]
+                                        << " otherPhi: " << otherPhi << " iPhi: " << iPhi << " otherEta: " << otherEta
+                                        << " iEta: " << iEta << "\n";
+
+              if ((otherEta > iEta && otherPhi > iPhi) || (otherEta > iEta && otherPhi == iPhi) ||
+                  (otherEta == iEta && otherPhi > iPhi))
```

Comment:
IIUC, ```otherEta == iEta && otherPhi == iPhi``` will never pass.
It seems easier then to put an outside ``` if (!(otherEta == iEta && otherPhi == iPhi) )``` and then, inside its scope this 3 comparisons condition is the same as ```otherEta >= iEta && otherPhi >= iPhi```, which is much easier to read.

Reply:
Done. Thanks! Indeed it looks much readable.